### ISSUE TITLE
Why GPT2 train loss and topK accuracy both decrease?

### ISSUE THREAD

**Issue by lx-kika** [huggingface/transformers#3523]
# ❓ Questions & Help

## Details

Hi,

I use GPT2 LMHeadModel training from scratch. The training loss decreases, however, I test it on the same train dataset, I get top3 accuracy decreasing as well. Moreover, is it normal that my top3 accuracy goes down sharply like from 0.2 to 0.05 only one or two epochs? Because it seems to be stable when it converges. Did anyone meet the same problem?

![loss](https://user-images.githubusercontent.com/62126666/77870461-14441380-7274-11ea-9406-c8c56bb8143f.jpg)

**A link to original question on Stack Overflow**:

**Comment by lx-kika** [comment]
I found a mistake that in GPT2LMHeadModel, the label is shifted, however, I shift it again when preparing the batch.
